qadence/backends/pulser/backend.py

```diff
@@ -235,13 +236,13 @@ def run_dm(
     ) -> list:
         vals = to_list_of_dicts(param_values)
         noise_probs = noise.options.get("noise_probs", None)
-        if noise_probs is None:
-            KeyError(f"A range of noise probabilies should be passed. Got {noise_probs}.")
+        if noise_probs is None or not isinstance(noise_probs, Iterable):
```

Comment:
i thought noise is a dataclass?

Reply:
ah, `noise_probs` sorry. are there any requirements for the shape of the noise?

Reply:
What do you mean by "shape of the noise" ?

Reply:
On second thoughts, `noise_probs` might either be a single float value or a container of floats. Will rectify this.